### DISCUSSION TITLE
query-index for preview purposes

### DISCUSSION THREAD

**Discussion by honstar** [adobe/helix-home#342] [Ideas]
For content collection use-cases on adobe.com, we consider using the OOTB Franklin indexing capabilities. The idea is to define a query-index that indexes HTML content from various pages, effectively building up a data source backed by an excel sheet. The OOTB rendition as JSON would be a perfect fit for client-side components to consume and render HTML content, preprocessed by custom business logic in terms of filtering, ordering etc.

In order to allow for previewing these assembled content collections for authors, it'd be great if the query-index would support indexing content without the need for explicit publishing. This preview query-index sheet could even be a separate excel file altogether to minimize potential conflicts with the production one.

I could imagine either tying the index trigger to a word document save event. Alternatively, an explicit sidekick trigger would also work I think.

**Comment by tripodsan** [comment]
> I could imagine either tying the index trigger to a word document save event. Alternatively, an explicit sidekick trigger would also work I think.

that is not realistic. if anything, it would be a index that is updated on preview, indexing the .page content.

**Reply by honstar** [reply]
Right, that makes sense 👍

**Comment by 3ch023** [comment]
@honstar speaking of a sidekick trigger, we could have a custom sidekick button that would post a request to add the page to the index. As far as I saw, API allows adding pages to the index 'manually'. But there might be problems with the fact that the page is not published, so needs a bit deeper investigation.

**Reply by honstar** [reply]
@dominique-pfister, can you let us know what the expected outcome/response is? Here's the use-case:
- create a new page to be indexed
- instead of publishing to add its content to the index, manually send a POST request to the helix index admin API

The idea is to be able to preview indexed content on `hlx.page` before it goes live.

**Reply by dominique-pfister** [reply]
The index admin API fetches HTML from `https://<ref>--<repo>--<owner>.hlx.live/<path>`, so if the page isn't there, or doesn't return a HTTP status code 200, nothing will be indexed.

**Reply by honstar** [reply]
Ok, thanks for the quick confirmation @dominique-pfister 👍

**Comment by honstar** [comment]
@tripodsan @dominique-pfister : I'd like to follow-up on this feature proposal and would like to understand if this would be something the core team would consider implementing? If yes, I'd be happy to contribute.

**Reply by tripodsan** [reply]
currently there is not much demand for this feature, and implementing it _correctly_ would require a major architecture change. if possible it should be implemented on the project.

**Comment by honstar** [comment]
cc @amol-anand
